modin/pandas/dataframe.py

```diff
@@ -179,6 +179,26 @@ def _init(
             elif is_dict_like(data) and not isinstance(
                 data, (pandas.Series, Series, pandas.DataFrame, DataFrame)
             ):
+                if columns is not None:
+                    data = {key: value for key, value in data.items() if key in columns}
+
+                if len(data) and all(isinstance(v, Series) for v in data.values()):
+                    from .general import concat
+
+                    new_qc = concat(
+                        data.values(), axis=1, keys=data.keys()
+                    )._query_compiler
+
+                    if dtype is not None:
+                        new_qc = new_qc.astype({col: dtype for col in new_qc.columns})
```

Comment:
Why not just?
```suggestion
                        new_qc = new_qc.astype(dtype)
```

Reply:
query compiler's API doesn't support that. It only accepts a mapping dictionary as `dtype` argument
https://github.com/modin-project/modin/blob/808b54ec79e47a6d9a1bbff8db943d2ae47b17e6/modin/core/storage_formats/base/query_compiler.py#L1415-L1418

Reply:
By the way, why do we work at the query compiler level? `concat` function returns a dataframe, why not work with it further?

Reply:
That's a safety catch for a case if some of the used methods would call `pd.DataFrame({col: pd.Series()})` to not end up in a recursive loop.